cmd/compact.go

```diff
@@ -0,0 +1,93 @@
+// Copyright (c) 2018 SAP SE or an SAP affiliate company. All rights reserved. This file is licensed under the Apache Software License, v. 2 except as noted otherwise in the LICENSE file.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cmd
+
+import (
+	"context"
+	"fmt"
+	"path/filepath"
+
+	"github.com/gardener/etcd-backup-restore/pkg/compactor"
+	"github.com/gardener/etcd-backup-restore/pkg/snapstore"
+	brtypes "github.com/gardener/etcd-backup-restore/pkg/types"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+	"go.etcd.io/etcd/pkg/types"
+)
+
+// NewCompactCommand compacts the ETCD instance
+func NewCompactCommand(ctx context.Context) *cobra.Command {
+	opts := newCompactOptions()
+	// compactCmd represents the restore command
+	compactCmd := &cobra.Command{
+		Use:   "compact",
+		Short: "compacts an etcd member data directory from snapshots",
+		Long:  fmt.Sprintf(`Compacts an etcd member data directory from existing backup stored in snapshot store.`),
+		Run: func(cmd *cobra.Command, args []string) {
```

Comment:
Should we do it? It would involve passing of arguments and returning many variables. We are already calling specialized functions in these lines

Reply:
It is not a hard opinion, but I am a bit worried about copying with variation of loading initial configuration, snapstore, listing backup etc. I think it might still be worth it especially when you take it with the the modularization of the code in pkg/compactor/compactor.go. WDYT?

Reply:
Updated